e2e/test_operator_maintenance_mode/operator_maintenance_mode_test.go

```diff
@@ -0,0 +1,163 @@
+				}
+
+				if processGroup.ProcessGroupID == fixtures.GetProcessGroupID(failingStoragePod) {
+					faultDomain = processGroup.FaultDomain
+				}
+			}
+
+			// Set the maintenance mode for 4 minutes.
+			fdbCluster.RunFdbCliCommandInOperator(fmt.Sprintf("maintenance on %s 240", faultDomain), false, 60)
+
+			// Set this Pod as unschedulable to keep it pending.
+			Expect(fdbCluster.SetPodAsUnschedulable(failingStoragePod)).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			// Make sure that the quota is deleted and new PVCs can be created.
+			Expect(fdbCluster.ClearBuggifyNoSchedule(true)).NotTo(HaveOccurred())
+			// Reset the maintenance mode
+			fdbCluster.RunFdbCliCommandInOperator("maintenance off", false, 60)
+		})
+
+		When("the Pod comes back before the maintenance mode times out", func() {
+			It("should not set the team tracker status to unhealthy", func() {
+				// Make sure the team tracker status shows healthy for the failed Pod and the maintenance zone is set.
+				Consistently(func(g Gomega) fdbv1beta2.FaultDomain {
+					status := fdbCluster.GetStatus()
+
+					for _, tracker := range status.Cluster.Data.TeamTrackers {
+						log.Println(tracker.State.Name, ":", tracker.State.Healthy)
+						g.Expect(tracker.State.Healthy).To(BeTrue())
+					}
+
+					log.Println("Maintenance Zone:", status.Cluster.MaintenanceZone)
+					return status.Cluster.MaintenanceZone
+				}).WithTimeout(2 * time.Minute).WithPolling(2 * time.Second).Should(Equal(faultDomain))
+			})
+		})
+
+		When("the maintenance mode times out", func() {
+			It("should update the team tracker status to unhealthy", func() {
+				// Make sure the team tracker status shows healthy for the failed Pod and the maintenance zone is set.
+				Consistently(func(g Gomega) fdbv1beta2.FaultDomain {
+					status := fdbCluster.GetStatus()
+
+					for _, tracker := range status.Cluster.Data.TeamTrackers {
+						g.Expect(tracker.State.Healthy).To(BeTrue())
+					}
+
+					return status.Cluster.MaintenanceZone
+				}).WithTimeout(2 * time.Minute).WithPolling(2 * time.Second).Should(Equal(faultDomain))
+
+				log.Println("Wait until maintenance mode times out")
+				// Wait until the maintenance zone is reset
+				Eventually(func(g Gomega) fdbv1beta2.FaultDomain {
+					return fdbCluster.GetStatus().Cluster.MaintenanceZone
+				}).WithTimeout(10 * time.Minute).WithPolling(2 * time.Second).Should(Equal(fdbv1beta2.FaultDomain("")))
+
+				startTime := time.Now()
+				log.Println("Wait until failure is detected")
+				// We would expect that the team tracker gets unhealthy once the maintenance mode is timed out.
+				Eventually(func(g Gomega) fdbv1beta2.FaultDomain {
+					status := fdbCluster.GetStatus()
+					for _, tracker := range status.Cluster.Data.TeamTrackers {
+						g.Expect(tracker.State.Healthy).To(BeFalse())
+					}
+
+					return status.Cluster.MaintenanceZone
+				}).WithTimeout(10 * time.Minute).WithPolling(2 * time.Second).Should(Equal(fdbv1beta2.FaultDomain("")))
```

Comment:
```suggestion
				}).WithTimeout(2 * time.Minute).WithPolling(2 * time.Second).Should(Equal(fdbv1beta2.FaultDomain("")))
```

We could reduce this to 2 minutes (or whatever we think is a reasonable upper bound). In my tests it roughly took a bit more than 1min to update the team trackers, which would align with the 60 seconds failure detection time.